osfoffline/polling_osf_manager/polling_events.py

```diff
@@ -21,117 +21,150 @@ def run(self):
+        try:
+            old_file_path = ProperPath(self.old_path, is_dir=False)
+        except Exception:
+            # TODO: Narrow down this exception and do client side warnings
+            logging.exception('Exception caught: Invalid origin path for renamed file.')
+            return
+        try:
+            new_file_path = ProperPath(self.new_path, is_dir=False)
+        except Exception:
+            # TODO: Narrow down this exception and do client side warnings
+            logging.exception('Exception caught: Invalid target path for renamed folder.')
+            return
+
+        AlertHandler.info(new_file_path.name, AlertHandler.MODIFYING)
+        yield from _rename(old_file_path, new_file_path)
 
 
 class UpdateFile(PollingEvent):
     def __init__(self, path, download_url, osf_query):
-        super().__init__(path)
-        self.path = ProperPath(path, is_dir=False)
+        super().__init__()
+        self.path = path
         self.osf_query = osf_query
         self.download_url = download_url
-        assert isinstance(self.osf_query, OSFQuery)
-        assert isinstance(self.download_url, str)
 
     @asyncio.coroutine
     def run(self):
-        AlertHandler.info(self.path.name, AlertHandler.MODIFYING)
-        try:
-            yield from _download_file(self.path, self.download_url, self.osf_query)
-        except Exception as e:
-            logging.warning(e)
-            # FIXME: Improve message to user here
-            # AlertHandler.warn("File unable to be updated online")
+        if not isinstance(self.osf_query, OSFQuery):
+            logging.error('Update file query is not an instance of OSFQuery')
+            return
+        if not isinstance(self.download_url, str):
+            logging.error('Update file download_url is not a str.')
+            return
+        updated_file_path = ProperPath(self.path, is_dir=False)
+        AlertHandler.info(updated_file_path.name, AlertHandler.MODIFYING)
+        yield from _download_file(updated_file_path, self.download_url, self.osf_query)
 
 
 class DeleteFolder(PollingEvent):
     def __init__(self, path):
-        super().__init__(path)
-        self.path = ProperPath(path, is_dir=True)
+        super().__init__()
+        self.path = path
 
     @asyncio.coroutine
     def run(self):
+        try:
+            folder_to_delete = ProperPath(self.path, is_dir=True)
+        except Exception:
+            # TODO: Narrow down this exception and do client side warnings
+            logging.exception('Exception caught: Invalid source path for deleted folder.')
+            return
         # this works on systems that use file descriptors.
         # thus, linux, mac are supported.
         # todo: is windows supported??
         if shutil.rmtree.avoids_symlink_attacks:
-            AlertHandler.info(self.path.name, AlertHandler.DELETING)
+            AlertHandler.info(folder_to_delete.name, AlertHandler.DELETING)
             shutil.rmtree(
```

Comment:
Am I misunderstanding how `onerror=lambda a, b, c: logging.warning('local node not deleted because it does not exist.')` works? I can certainly wrap it, but I thought it would be redundant. That being said, I didn't do a lot of research into rmtree.